consensus/bor/bor.go

```diff
@@ -990,19 +986,10 @@ func (c *Bor) Finalize(config *chain.Config, header *types.Header, state *state.
 	txs types.Transactions, uncles []*types.Header, r types.Receipts, withdrawals []*types.Withdrawal,
 	chain consensus.ChainReader, syscall consensus.SystemCall, logger log.Logger,
 ) (types.Transactions, types.Receipts, error) {
-	var err error
-
 	headerNumber := header.Number.Uint64()
 
-	if withdrawals != nil {
-		// withdrawals != nil not required because withdrawals are not used
-		log.Warn("Bor does not support withdrawals", "number", headerNumber)
-	}
-
-	if header.WithdrawalsHash != nil {
-		header.WithdrawalsHash = nil
-
-		log.Warn("Bor does not support withdrawalHash", "number", headerNumber)
+	if withdrawals != nil || header.WithdrawalsHash != nil {
+		return nil, nil, consensus.ErrUnexpectedWithdrawals
```

Comment:
Should we return an error here? I had just changed the values here because it might be a potential attack vector where the attacker can break the consensus by sending withdrawals object. thoughts?